test/spec_utils.rb

```diff
@@ -354,6 +354,43 @@ def initialize(*)
     ]
   end
 
+  it "parses RFC 7239 Forwarded header" do
+    Rack::Utils.forwarded_values('for=3.4.5.6').must_equal({
+      :for => [ '3.4.5.6' ],
+    })
+
+    Rack::Utils.forwarded_values(';;;for=3.4.5.6,,').must_equal({
+      :for => [ '3.4.5.6' ],
+    })
+
+    Rack::Utils.forwarded_values('for=3.4.5.6').must_equal({
+      :for => [ '3.4.5.6' ],
+    })
+
+    Rack::Utils.forwarded_values('for =  3.4.5.6').must_equal({
+      :for => [ '3.4.5.6' ],
+    })
+
+    Rack::Utils.forwarded_values('for="3.4.5.6"').must_equal({
+      :for => [ '3.4.5.6' ],
+    })
+
+    Rack::Utils.forwarded_values('for=3.4.5.6;proto=https').must_equal({
+      :for   => [ '3.4.5.6' ],
+      :proto => [ 'https' ]
+    })
+
+    Rack::Utils.forwarded_values('for=3.4.5.6; proto=http, proto=https').must_equal({
+      :for   => [ '3.4.5.6' ],
+      :proto => [ 'http', 'https' ]
+    })
+
+    Rack::Utils.forwarded_values('for=3.4.5.6; proto=http, proto=https; for=1.2.3.4').must_equal({
+      :for   => [ '3.4.5.6', '1.2.3.4' ],
+      :proto => [ 'http', 'https' ]
+    })
+  end
+
```

Comment:
Looks like we're covering https://tools.ietf.org/html/rfc7239#section-7.1 👍 

Can't do much about multiple header testing since that falls on the Rack handler to coalesce.